tests/FSharp.Compiler.ComponentTests/EmittedIL/Attribute/Attribute.fs

```diff
@@ -0,0 +1,42 @@
+﻿namespace EmittedIL.RealInternalSignature
+
+open Xunit
+open FSharp.Test
+open FSharp.Test.Compiler
+
+module Attribute =
+
+    let verifyCompilation compilation =
```

Comment:
Could you please add test cases(s) with mutual recursion (either via `and` or via a `rec module`) between the attribute declaration, the type argument and the type it is being applied on?

Reply:
Do we have a particular point of failure we are looking for here? Off the top of my head this is what comes to mind:
(in a module rec)
```fsharp
type MyAttribute<^T>() =
    inherit Attribute()

[<MyAttribute<AnActualType>>]
let x = 1

type AnActualType = class end
```
(Note: it appears that in a rec module, the usage of an attribute on a binding needs to occur after the definition, but on a type it can occur before the definition.)
and maybe something like

(in a regular module)
```fsharp
[<MyAttribute<AnActualType>>]
type SomeClass(arg) =
    member _.AGoodProperty = arg

and MyAttribute<^T>() =
           inherit Attribute()

and type AnActualType = int
```

Reply:
Oh great call, both rec and and fail right now (made sure to check they pass when there are no type args and both do)

Reply:
Glad I could suggest a point of failure - I did not have a particular one in mind, I just know that rec modules are a common point of failure for new typechecking features.


Maybe also a type applying "itself" as the type argument to the attribute?

```
[<MyAttribute<SomeClass<SomeClass<SomeClass<MyAttribute>>>>>]
type SomeClass<'a>(arg) =

```

Reply:
I still need to figure out why the rec/and cases failed, but I implemented the self referencing case (great case btw, I know there must be some cases where it will be useful but this in particular made me laugh quite hard), and was surprised to find it working right out of the gate!